From: Kurt Garloff <garloff@suse.de>
Subject: Initialize readahead in vbd Q init code

The domU read performance is poor without readahead, so
better make sure we initialize this value.

Signed-off-by: Kurt Garloff <garloff@suse.de>

Index: linux-2.6.11/drivers/xen/blkfront/vbd.c
===================================================================
--- linux-2.6.11.orig/drivers/xen/blkfront/vbd.c
+++ linux-2.6.11/drivers/xen/blkfront/vbd.c
@@ -268,8 +268,11 @@ static struct gendisk *xlvbd_get_gendisk
             xlbd_blk_queue, BLKIF_MAX_SEGMENTS_PER_REQUEST);
 
         /* Make sure buffer addresses are sector-aligned. */
         blk_queue_dma_alignment(xlbd_blk_queue, 511);
+
+	/* Set readahead */
+	blk_queue_max_sectors(xlbd_blk_queue, 128);
     }
     gd->queue = xlbd_blk_queue;
 
     add_disk(gd);
